### PR TITLE
test(integration): initial version

### LINKED ISSUE #26: Create integration test
Happy path test: renders form when user is authenticated and project can be found

### PR DESCRIPTION
**Pull request by oscard0m**
This PR is meant to be used as guidance for implementing https://github.com/project-forms/project-forms.github.io/pull/28 by @gr2m as a way to get familiar with [MSW](https://mswjs.io/).

closes #26, closes #28

### CODE REVIEW

File: src/components/octokit-provider.js [hunk 1]
```diff
@@ -143,15 +146,18 @@ async function handleCodeInUrl(store, setAuthState, code) {
...
   try {
-    const response = await fetch(`${backendBaseUrl}/api/github/oauth/token`, {
-      method: "POST",
-      headers: {
-        "content-type": "application/json",
-      },
-      body: JSON.stringify({ code }),
-    });
+    const response = await fetch(
+      `${backendBaseUrl || location.origin}/api/github/oauth/token`,
```
Comment (oscard0m) [inline src/components/octokit-provider.js:152]: ⚠️

File: src/pages/{org}/{repo}/projects/{project_number}/issues/new.jsx [hunk 2]
```diff
@@ -132,7 +143,9 @@ export default function NewIssuePage() {
   /**
    * @type {import("..").Store<import("..").StoreData>}
    */
-  const store = createStore(currentPath);
+
+  const store =
+    process.env.NODE_ENV === "test" ? DEFAULT_STORE : createStore(currentPath);
```
Comment (oscard0m) [inline src/pages/{org}/{repo}/projects/{project_number}/issues/new.jsx:148]: ⚠️

File: src/test/intergration.test.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+const server = setupServer();
+const githubGraphQL = graphql.link("https://api.github.com/graphql");
+class ResizeObserver {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+}
+describe("App", () => {
+  beforeAll(() => {
+    window.ResizeObserver = ResizeObserver;
```
Comment (oscard0m) [inline src/test/intergration.test.jsx:19]: ⚠️

File: src/test/intergration.test.jsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            },
+          },
+        });
+      })
+    );
+
+    const { asFragment, getByText, debug } = render(
+      <ThemeProvider>
+        <BaseStyles>
+          <OctokitProvider location={mockLocation}>
```
Comment (oscard0m) [inline src/test/intergration.test.jsx:99]: ⚠️

File: src/pages/{org}/{repo}/projects/{project_number}/issues/new.jsx (hunk 2 above)
Comment (gr2m) [reply src/pages/{org}/{repo}/projects/{project_number}/issues/new.jsx:148]: We will replace this workaround with exposing the store API from the OctokitProvider in a follow up PR

File: src/test/intergration.test.jsx (hunk 3 above)
Comment (gr2m) [inline src/test/intergration.test.jsx:19]:
```suggestion
    // It seems like Primer components rely on the existence of `window.ResizeObserver` and it's not (properly) mocked by JS Dom
    window.ResizeObserver = ResizeObserver;
```

**Pull request merged**